### DISCUSSION TITLE
I need assistance upgrading from Laravel 8 to Laravel 9

### DISCUSSION THREAD

**Discussion by Ifriqiya** [laravel/framework#42039] [Q&A] [answered]
I'm getting this response on the very first step of trying to upgrade to L9.

"Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - illuminate/support[v5.6.0, ..., 5.8.x-dev] require php ^7.1.3 -> your php version (8.1.1) does not satisfy that requirement.
    - illuminate/support[v6.0.0, ..., v6.19.1] require php ^7.2 -> your php version (8.1.1) does not satisfy that requirement.    
    - illuminate/support[v7.0.0, ..., v7.28.4] require php ^7.2.5 -> your php version (8.1.1) does not satisfy that requirement.  
    - illuminate/support[v8.0.0, ..., v8.11.2] require php ^7.3 -> your php version (8.1.1) does not satisfy that requirement.    
    - Root composer.json requires nexmo/laravel ^2.4 -> satisfiable by nexmo/laravel[2.4.0, 2.4.1].
    - Conclusion: don't install laravel/framework v9.0.0-beta.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.0-beta.3 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.0-beta.4 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.0-beta.5 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.2 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.1.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.2.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.3.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.3.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.4.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.4.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.5.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.5.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.6.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.7.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.8.0 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.8.1 (conflict analysis result)
    - Conclusion: don't install laravel/framework v9.0.0-beta.1 (conflict analysis result)
    - nexmo/laravel[2.4.0, ..., 2.4.1] require illuminate/support ^5.2|^6.0|^7.0|^8.0 -> satisfiable by illuminate/support[v5.2.0, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev].
    - Only one of these can be installed: illuminate/support[v5.2.0, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev], laravel/framework[v9.0.0-beta.1, ..., 9.x-dev]. laravel/framework replaces illuminate/support and thus cannot coexist with it.
    - Root composer.json requires laravel/framework ^9.0 -> satisfiable by laravel/framework[v9.0.0-beta.1, ..., 9.x-dev].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions."

This is my composer.json:

{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The Laravel Framework.",
    "keywords": ["framework", "laravel"],
    "license": "MIT",
    "require": {
        "php": "^8.1",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "inertiajs/inertia-laravel": "^0.5.4",
        "lab404/laravel-impersonate": "^1.7",
        "laravel/framework": "^9.0",
        "laravel/jetstream": "^2.4",
        "laravel/nexmo-notification-channel": "^2.5",
        "laravel/sanctum": "^2.11",
        "laravel/telescope": "^4.6",
        "laravel/tinker": "^2.5",
        "nexmo/laravel": "^2.4",
        "propaganistas/laravel-phone": "^4.3",
        "pusher/pusher-php-server": "^7.0",
        "spatie/laravel-permission": "^5.3",
        "tightenco/ziggy": "^1.0",
        "wildbit/swiftmailer-postmark": "^3.3"
    },
    "require-dev": {
        "spatie/laravel-ignition": "^1.0",
        "fakerphp/faker": "^1.9.1",
        "laravel/dusk": "^6.19",
        "laravel/sail": "^1.0.1",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^6.1",
        "phpunit/phpunit": "^9.5.10"
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ]
    },
    "extra": {
        "laravel": {
            "dont-discover": [
                "laravel/telescope"
            ]
        }
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}

How can I resolve this?

**Comment by PaolaRuby** [comment]
https://laravelshift.com/

**Reply by Ifriqiya** [reply]
I'm aware of laravel shift, thank you. But I'd like to do this manually. I want to know what the problem is and how to resolve it directly.

**Reply by gcjbr** [reply]
It's funny because Shift created me this issue

**Comment by hoczeus** [comment, marked as answer]
The interesting line is this one:
- nexmo/laravel[2.4.0, ..., 2.4.1] require illuminate/support ^5.2|^6.0|^7.0|^8.0 -> satisfiable by illuminate/support[v5.2.0, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev].

That means nexmo/laravel needs laravel till 8. One Solution is updating nexmo/laravel to latest version, as I can see 3.0
But in the nexmo/laravel repo there is an Open Issue with Title. "Support Laravel 9 [#73](https://github.com/Nexmo/nexmo-laravel/issues/73)"
It seams that nexmo/laravel is not supporting Laravel 9 at the moment. Do you need it?

**Reply by Ifriqiya** [reply]
Hi, thanks for this, you're correct indeed, I replaced nexmo/laravel with vonage/laravel and 'composer update --no-scripts worked. Thanks again.

**Reply by volkankaban** [reply]
Don't forget to remove fruitcake/php-cors

**Reply by MuscledGeek** [reply]
I was upgrading v9 to v10, removing fruitcake/php-cors worked, thanks.

**Comment by ransh-f** [comment]
"spatie/laravel-permission": "^5.3", spatie version at list "spatie/laravel-permission": "^5.7" for laravel 9

**Reply by josephajibodu** [reply]
This worked for me. Thanks
